src/idl_gen_fbs.cpp

```diff
@@ -136,8 +140,28 @@ std::string GenerateFBS(const Parser &parser, const std::string &file_name) {
         GenComment(field.doc_comment, &schema, nullptr, "  ");
         schema += "  " + field.name + ":" + GenType(field.value.type);
         if (field.value.constant != "0") schema += " = " + field.value.constant;
-        if (field.IsRequired()) schema += " (required)";
-        if (field.key) schema += " (key)";
+        bool metadata = false;
+        if (field.IsRequired()) {
+          metadata = true;
```

Comment:
I would get rid of the metadata checks and just fill in a vector<std::string> of all the attributes we need to apply. Then if that vector is !empty(), you can just write a simple loop to append "(" ... all the vector items ... ")".

Reply:
Done!